lib/matplotlib/patheffects.py

```diff
@@ -373,3 +375,173 @@ def draw_path(self, renderer, gc, tpath, affine, rgbFace):
         if clip_path:
             self.patch.set_clip_path(*clip_path)
         self.patch.draw(renderer)
+
+
+class TickedStroke(AbstractPathEffect):
+    """
+    A line based PathEffect which draws a path with a ticked style.
+    This line style is frequently used to represent constraints in
+    optimization.  The ticks indicate that one side of the line is
+    invalid.
+
+    The spacing, length, and angle of ticks can be controlled.
+
+    This line style is sometimes referred to as a hatched line, but
+    in matplotlib, hatching typically refers to filling an area with
+    a line pattern.
+
+    """
+
+    def __init__(self, offset=(0, 0),
+                 spacing=10.0, angle=45.0, length=np.sqrt(2),
+                 flip=False, **kwargs):
+        """
+        Parameters
+        ----------
+        offset : pair of floats
+            The offset to apply to the path, in points.
+        spacing : float
+            The spacing between ticks in points.
+            Default is 10.0.
+        angle : float
+            The angle between the path and the tick in degrees.  The angle
+            is measured as if you were an ant walking along the curve, with
+            zero degrees pointing directly behind you, 90 to your left, 180
+            straight ahead, and 270 to your right.  If the curve was
```

Comment:
Ha, ok, that's a bit colloquial, but its certainly clear.  OTOH, is that actually the convention?  I'd have instinctively thought 0 was the along-curve direction and 90 to the left, 180 behind and 270 to right.    I don't know that I'd block over this, but is there a good reason for this being clockwise and staring from aft?

Reply:
Sorry if the prose documentation is overkill -- I struggled to find a medium between what I had and what I ended up with.  Suggestions appreciated.

I agree with your comments about the angle convention.  I hadn't really been caring about the convention when I developed this.  I see them as arbitrary, with the ability to set different angles a way of making styles unique.  My main concern was that 'easy' angles resulted in the correct side for use with constraints in an optimization sense.

Commit e094b4497d changes it to what you expect.  Two negations are needed.  First to flip the CW to CCW (-theta) and second, to construct the unit vector along the curve in the direction you're headed (instead of backwards).

One could make an argument to follow a CW compass heading convention vs. a trig convention, but that would mean the default would need to be -45, and nobody wants that.